poly/promise/promise.go

```diff
@@ -0,0 +1,1355 @@
+package promise
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"io"
+	"reflect"
+	"strconv"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/anon"
+	"github.com/dedis/crypto/config"
+	"github.com/dedis/crypto/poly"
+	"github.com/dedis/crypto/proof"
+	"github.com/dedis/crypto/random"
+)
+
+/* This package implements the Promise cryptographic primitive, which is based
```

Comment:
Make sure this overview text gets included in a sensible way in godoc-generated documentation.  It might be appropriate for the "package-level" documentation, for example, which should go just above the 'package' statement if so.